WordPress/Classes/ViewRelated/Media/SiteMedia/Views/MediaStorageDetailsView.swift

```diff
@@ -0,0 +1,482 @@
+import Foundation
+import SwiftUI
+import SVProgressHUD
+import WordPressAPI
+import WordPressCore
+import WordPressData
+
+struct MediaStorageDetailsView: View {
+    @State private var purchase: WebPurchase?
+    @ObservedObject private var viewModel: MediaStorageDetailsViewModel
+
+    @Environment(\.dismiss) private var dismiss
+
+    init(viewModel: MediaStorageDetailsViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section {
+                    UsageView(usage: viewModel.usage)
+                } footer: {
+                    if viewModel.isLoading {
+                        HStack {
+                            Spacer()
+                            ProgressView()
+                            Spacer()
+                        }
+                        .padding(.top, 8)
+                    }
+                }
+
+                ForEach(viewModel.actions, id: \.kind) { action in
+                    actionSection(action)
+                }
+            }
+            .navigationTitle(Strings.title)
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .navigationBarLeading) {
+                    if #available(iOS 26.0, *) {
+                        Button(role: .cancel) {
+                            dismiss()
+                        }
+                    } else {
+                        Button(role: .cancel) {
+                            dismiss()
+                        } label: {
+                            Image(systemName: "xmark")
+                        }
+                    }
+                }
+            }
+            .sheet(item: $purchase) { purchase in
+                WebPurchaseView(url: purchase.url, customTitle: purchase.title) { _ in
+                    SVProgressHUD.showSuccess(withStatus: purchase.successMessage)
+
+                    self.purchase = nil
+                    Task {
+                        await viewModel.refresh()
+                    }
+                }
+                .preferredColorScheme(.light)
+                .onDisappear {
+                    self.purchase = nil
+                }
+            }
+            .task {
+                await viewModel.refresh()
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func actionSection(_ action: Action) -> some View {
+        switch action.kind {
+        case .cleanUp:
+            ActionSection(action: action) {
+                // TODO: Implement a view that shows unattached media (order by file size)
```

Comment:
Note: I tested and noticed it was not implemented. Turns out there was a respective TODO.